bls/bls_fuzz_test.go

```diff
@@ -0,0 +1,185 @@
+package bls
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/trailofbits/go-fuzz-utils"
+)
+
+func GetTypeProvider(data []byte) (*go_fuzz_utils.TypeProvider, error) {
+	tp, err := go_fuzz_utils.NewTypeProvider(data)
+	if err != nil {
+		return nil, err
+	}
+
+	// Prevent nil fields.
+	err = tp.SetParamsBiases(0, 0, 0, 0)
+	if err != nil {
+		return nil, err
+	}
+	return tp, nil
+}
+
+func FuzzPublicKeyFromBytes(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		PublicKeyFromBytes(data)
+	})
+}
+
+func FuzzPublicKeyFromBytesValidLength(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		tp, err := GetTypeProvider(data)
+		if err != nil {
+			return
+		}
```

Comment:
```suggestion
		require.NoError(t, err)
```

you can use this everywhere :)

Reply:
We can't require no errors in those spots. It will return an error if there's not enough data to build the type. Technically it's not a test failure. It needs to bail and try again.

See: https://github.com/trailofbits/go-fuzz-utils#setup

> not enough data was supplied, exit gracefully for the next fuzzing iteration

Reply:
oh, right!